lib/Controller/EndpointController.php

```diff
@@ -62,17 +62,25 @@ public function __construct(string $appName,
 
 
 	/**
+	 * @param string $rootPassword
+	 *
 	 * @return DataResponse
 	 */
-	public function Reboot(): DataResponse {
+	public function Reboot($rootPassword): DataResponse {
+		var_dump($rootPassword);
+		exit;
 		$result = $this->os->reboot();
 		return new DataResponse();
 	}
 
 	/**
+	 * @param string $rootPassword
+	 *
 	 * @return DataResponse
```

Comment:
`@PasswordConfirmationRequired` is missing, otherwise you can directly call the API without the password

Reply:
@nickvergessen This is the root password of the server, not the user's password.

Reply:
And if you call the API without the password, it will simply fail, because reboot will require the root password.

Reply:
Ah okay, misunderstood